ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx

```diff
@@ -69,71 +70,95 @@ function setVisibleEdges(edges) {
+                extraneousEgressNode?.setVisible(true);
+            } else if (networkPolicyState === 'egress') {
+                // if the node has egress policies from policy graph, show extraneous ingress node
+                extraneousIngressNode?.setVisible(true);
+            } else if (networkPolicyState === 'none') {
+                // if the node has no policies, show both extraneous ingress and egress nodes
+                extraneousEgressNode?.setVisible(true);
+                extraneousIngressNode?.setVisible(true);
+            }
         }
     }
-}
 
-function setExtraneousNodes(controller, detailId) {
-    if (!detailId) {
-        // if there is no selected node, check if extraneous nodes exist and clear them
-        const extraneousIngressNode = controller.getNodeById('extraneous-ingress');
+    function hideExtraneousNodes() {
+        // if there is no selected node, check if extraneous nodes exist and remove them
+        const extraneousIngressNode = controller.getElementById('extraneous-ingress');
         if (extraneousIngressNode) {
-            controller.removeElement(extraneousIngressNode);
+            extraneousIngressNode.setVisible(false);
         }
-        const extraneousEgressNode = controller.getNodeById('extraneous-egress');
+        const extraneousEgressNode = controller.getElementById('extraneous-egress');
         if (extraneousEgressNode) {
-            controller.removeElement(extraneousEgressNode);
+            extraneousEgressNode.setVisible(false);
         }
-    } else {
+    }
+
+    function setExtraneousEdges() {
         const currentModel = controller.toModel();
-        const { extraneousEgressNode, extraneousIngressNode } = createExtraneousNodes();
-        // else if there is a selected node, create a node to collect extraneous flows
+        const extraneousIngressNode = controller.getElementById('extraneous-ingress');
+        const extraneousEgressNode = controller.getElementById('extraneous-egress');
+        const { extraneousEgressEdge, extraneousIngressEdge } = createExtraneousEdges(detailId);
         const selectedNode = controller.getNodeById(detailId);
-        const { networkPolicyState } = selectedNode?.data || {};
-        if (networkPolicyState === 'ingress') {
-            // if the node has ingress policies from policy graph, create extraneous egress node
-            currentModel.nodes.push(extraneousEgressNode);
-        } else if (networkPolicyState === 'egress') {
-            // if the node has egress policies from policy graph, create extraneous ingress node
-            currentModel.nodes.push(extraneousIngressNode);
-        } else if (networkPolicyState === 'none') {
-            // if the node has no policies, create both extraneous ingress and egress nodes
-            currentModel.nodes.push(extraneousEgressNode);
-            currentModel.nodes.push(extraneousIngressNode);
+        const { data } = selectedNode || {};
```

Comment:
also getting a ts error for this line -- not sure why the controller doesn't return a `NodeModel` and instead returns `Node<NodeModel, any>`